src/writer.cpp

```diff
@@ -1,12 +1,14 @@
 #include "writer.h"
 #include <iostream>
+#include "encoder.h"
 #include "version_number.h"
 
 namespace datadog {
 namespace opentracing {
 
+Writer::Writer() : encoder_(std::unique_ptr<HttpEncoder>{new AgentHttpEncoder{}}) {}
```

Comment:
Could we assign this in the class def?

Reply:
This is the one bit I wouldn't want to put in there.
It's the point where we set the bits we control for the writer that is extended by something else.